### PR TITLE
Write Tests for Project 3 Sentiment Analysis

### PR DESCRIPTION
**Pull request by cel243**
I used mockito and Junit to test the sentiment class from project 3.

### CODE REVIEW

File: portfolio/src/test/java/com/google/sps/TestSentiment.java [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+import com.google.cloud.language.v1.ClassifyTextRequest;
+import com.google.cloud.language.v1.ClassifyTextResponse;
+import com.google.cloud.language.v1.ClassificationCategory;
+import com.google.api.gax.rpc.InvalidArgumentException;
+import java.lang.Throwable;
+import com.google.api.gax.grpc.GrpcStatusCode;
+import io.grpc.Status;
+
+@RunWith(JUnit4.class)
+public final class TestSentiment {
```
Comment (yutaow) [inline portfolio/src/test/java/com/google/sps/TestSentiment.java:34]: nit: usually each XX.java class has a unittest named as XXTest.java.

Comment (cel243) [reply portfolio/src/test/java/com/google/sps/TestSentiment.java]: Thanks! The file is now named "SentimentTest"

File: portfolio/src/test/java/com/google/sps/SentimentTest.java [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+    when(mockClassificationCategoryOne.getName()).thenReturn(CATEGORY_NAME_ONE);
+    when(mockClassificationCategoryTwo.getName()).thenReturn(CATEGORY_NAME_TWO);
+
+    ENTITY_ONE_METADATA_MAP.put("wikipedia_url", ENTITY_ONE_LINK);
+    ENTITY_TWO_METADATA_MAP.put("wikipedia_url", ENTITY_TWO_LINK);
+  }
+
+  @Test 
+  public void negativeSentiment() {
```
Comment (davidgaskins) [inline portfolio/src/test/java/com/google/sps/SentimentTest.java:113]: could you try to incorporate this comment into the test name? then you can get ride of this comment

File: portfolio/src/test/java/com/google/sps/SentimentTest.java [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+    // when the sentiment analyzer returns -1, then the SentimentAnalyzer class
+    // classifies the text as NEGATIVE.
+
+    when(mockSentiment.getScore()).thenReturn((float) -1);
```
Comment (davidgaskins) [inline portfolio/src/test/java/com/google/sps/SentimentTest.java:117]: I think you can do this more succinctly -1f?

File: portfolio/src/test/java/com/google/sps/SentimentTest.java (hunk 2 above)
Comment (davidgaskins) [reply portfolio/src/test/java/com/google/sps/SentimentTest.java:113]: here and below

File: portfolio/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 4]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (davidgaskins) [inline portfolio/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: new line please :)

**Pull request merged**